fsspec/implementations/http.py

```diff
@@ -671,13 +671,19 @@ async def _file_size(url, session=None, size_policy="head", **kwargs):
     else:
         raise TypeError('size_policy must be "head" or "get", got %s' "" % size_policy)
     async with r:
-        # TODO:
-        #  recognise lack of 'Accept-Ranges', or  'Accept-Ranges': 'none' (not 'bytes')
-        #  to mean streaming only, no random access => return None
-        if "Content-Length" in r.headers:
-            return int(r.headers["Content-Length"])
-        elif "Content-Range" in r.headers:
-            return int(r.headers["Content-Range"].split("/")[1])
+        try:
+            r.raise_for_status()
+
+            # TODO:
+            #  recognise lack of 'Accept-Ranges', or  'Accept-Ranges': 'none' (not 'bytes')
+            #  to mean streaming only, no random access => return None
+            if "Content-Length" in r.headers:
+                return int(r.headers["Content-Length"])
+            elif "Content-Range" in r.headers:
+                return int(r.headers["Content-Range"].split("/")[1])
+        except:
```

Comment:
I think flake will still complain here; indeed it would be better to know what sort of exception we expect. `aiohttp.ClientError` ?

Reply:
Indeed. Fixed it and ran the pre-commit hook locally. Should be all good now.

Reply:
I see you've used a `requests` error - but these should come from aiohttp, no? I don't think the two are compatible. Your test does pass somehow... can you please ensure that this block actually activates during your test?

Reply:
Fixed it. Was mistakenly thinking it was a reqeusts.session that was used.